src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs

```diff
@@ -0,0 +1,521 @@
+                    configuration.CachedConnections?.GetMostRecentConnection();
+                this.ServerComboBox.Text = connectionInfo?.ServerUri == null ? string.Empty : connectionInfo.ServerUri.AbsoluteUri;
+                InitializeView();
+            }
+        }
+
+        /// <summary>
+        /// Adds the currently selected connection to the configuration so it is persisted
+        /// in the MRU cache as well as the auto-startup connection
+        /// </summary>
+        /// <param name="configuration"></param>
+        public void UpdateConfigFromSelections(ExtensionConfiguration configuration)
+        {
+            var connection = GetConnectionFromTreeView();
+            if (connection != null)
+            {
+                configuration.CachedConnections?.AddToCache(connection);
+                configuration.SavedConnection = connection;
+            }
+        }
+
+        /// <summary>
+        /// For this control we want SaveAndClose to be enabled if any team project
+        /// is selected, regardless of whether it differs from the current configuration.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public bool IsConfigurationChanged()
+        {
+            var connection = GetConnectionFromTreeView();
+            return connection?.IsPopulated == true;
+        }
+        #endregion
+
+        #region state change button handlers
+        /// <summary>
+        /// Logs the user in and moves to editing server screen.
+        /// Forces a configuration change to the saved connection so the server URL is set, 
+        /// but the team project and team are null
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void NextButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (InteractionAllowed)
+            {
+                if (Uri.IsWellFormedUriString(ServerComboBox.Text, UriKind.Absolute))
+                {
+                    var serverUri = ToUri(ServerComboBox.Text);
+
+                    // block clicking "next" until login request is done
+                    ToggleLoading(true);
+                    await AzureDevOps.HandleLoginAsync(true, serverUri).ConfigureAwait(false);
+
+                    if (AzureDevOps.ConnectedToAzureDevOps)
+                    {
+                        AzureDevOps.Configuration.SavedConnection = FileIssueHelpers.CreateConnectionInfo(serverUri, null, null);
+                        ChangeStates(States.EditingServer);
+                    }
+                    else
+                    {
+                        ToggleLoading(false);
+                        ServerComboBox.Focus();
+                    }
+                }
+                else
+                {
+                    //MessageDialog.Show("URL format is not valid. Example URL: https://accountname.visualstudio.com");
```

Comment:
Do we want to do something in here? If not we should get rid of the else.

Reply:
Yes, I still need to port over MessageDialog